dto.go

```diff
@@ -0,0 +1,75 @@
+package basm
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/mailru/easyjson"
+)
+
+func marshal(v any) ([]byte, error) {
+	easy, ok := v.(easyjson.Marshaler)
+	if !ok {
+		return nil, errors.New("value does not implement easyjson.Marshaler")
+	}
+	return easyjson.Marshal(easy)
+}
+
+func unmarshal(data []byte, v any) error {
+	easy, ok := v.(easyjson.Unmarshaler)
+	if !ok {
+		return errors.New("value does not implement easyjson.Unmarshaler")
+	}
+	return easyjson.Unmarshal(data, easy)
+}
```

Comment:
we specify some in-package ser/deser functions for json across the wasm boundary. Here we use easyjson for now (which uses code generation to skip a lot of the use of the reflect package in `encoding/json`). Structs are updated via `make generate`